osde2e/managed_cluster_validating_webhooks_test.go

```diff
@@ -5,11 +5,606 @@ package osde2etests
+			Expect(apierrors.IsForbidden(err)).To(BeTrue())
+
+			err = userk8s.Create(ctx, withNamespace(pod, privilegedNamespace))
+			Expect(apierrors.IsForbidden(err)).To(BeTrue())
+
+			err = userk8s.Create(ctx, withNamespace(pod, unprivilegedNamespace))
+			Expect(apierrors.IsForbidden(err)).To(BeTrue())
+		}, SpecTimeout(createPodWaitDuration.Seconds()+deletePodWaitDuration.Seconds()))
+
+		It("allows cluster-admin to schedule pods onto master/infra nodes", func(ctx context.Context) {
+			sa := &v1.ServiceAccount{}
+
+			err := client.Get(ctx, saName, namespaceName, sa)
+
+			if err == nil {
+				err = client.Delete(ctx, sa)
+				Expect(err).ToNot(HaveOccurred(), "Failed to delete existing Service Account")
+			}
+
+			sa = &v1.ServiceAccount{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      saName,
+					Namespace: namespaceName,
+				},
+			}
+			err = client.Create(ctx, sa)
+			Expect(err).ShouldNot(HaveOccurred(), "Unable to create service account")
+
+			pod = withNamespace(pod, privilegedNamespace)
+			err = client.Create(ctx, pod)
+			Expect(err).NotTo(HaveOccurred())
+			err = client.Delete(ctx, pod)
+			Expect(err).NotTo(HaveOccurred())
+		}, SpecTimeout(createPodWaitDuration.Seconds()+deletePodWaitDuration.Seconds()))
+
+		It("prevents operators workloads from being scheduled on worker or master nodes", func(ctx context.Context) {
+			operators := map[string]string{
+				"cloud-ingress-operator":          "openshift-cloud-ingress-operator",
+				"configure-alertmanager-operator": "openshift-monitoring",
+				"custom-domains-operator":         "openshift-custom-domains-operator",
+				"managed-velero-operator":         "openshift-velero",
+				"must-gather-operator":            "openshift-must-gather-operator",
+				"osd-metrics-exporter":            "openshift-osd-metrics",
+				"rbac-permissions-operator":       "openshift-rbac-permissions",
+			}
+
+			var podList v1.PodList
+			err = client.WithNamespace(metav1.NamespaceAll).List(ctx, &podList)
+			Expect(err).ShouldNot(HaveOccurred(), "unable to list pods")
+			Expect(len(podList.Items)).To(BeNumerically(">", 0), "found no pods")
+
+			var nodeList v1.NodeList
+			selectInfraNodes := resources.WithLabelSelector(labels.FormatLabels(map[string]string{"node-role.kubernetes.io": "infra"}))
+			err = client.List(ctx, &nodeList, selectInfraNodes)
+			Expect(err).ShouldNot(HaveOccurred(), "unable to list infra nodes ")
+
+			nodeNames := []string{}
+			for _, node := range nodeList.Items {
+				nodeNames = append(nodeNames, node.GetName())
+			}
+
+			violators := []string{}
+			for _, pod := range podList.Items {
+				for operator, namespace := range operators {
+					if pod.GetNamespace() != namespace {
+						continue
+					}
+					if strings.HasPrefix(pod.GetName(), operator) && !strings.HasPrefix(pod.GetName(), operator+"-registry") {
+						if !slice.ContainsString(nodeNames, pod.Spec.NodeName, nil) {
```

Comment:
https://pkg.go.dev/slices#Contains

swap this out to drop the kubectl dep for stdlib